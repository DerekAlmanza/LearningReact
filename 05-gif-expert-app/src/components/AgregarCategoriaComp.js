import React, { useState } from 'react'

export const AgregarCategoria = () => {

	const [value, setValue] = useState('Escribe una categoría a agregar');

	/**
	 * Función que se encarga de recibir el valor del input
	 */
	const cambiarValor = (e) => {
		setValue(e.target.value);
	}

	/**
	 * Función del form realizada para que no se actualice el form automáticamente.
	 */
	const subirInfoManualmente = (e) => {
		e.preventDefault();
	}

    return (
        <form onSubmit={subirInfoManualmente}>
        	<h3> Agregar Categoria </h3>
        	<input 
				type='text' 
				placeholder={value}
				onChange={cambiarValor}
			/> 
        </form>
    )
}
