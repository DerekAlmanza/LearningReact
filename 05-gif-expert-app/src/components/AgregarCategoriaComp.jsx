import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AgregarCategoria = ({setCategorias, categorias}) => {

	const [value, setValue] = useState('');

	/**
	 * Función que se encarga de recibir el valor del input
	 */
	const cambiarValor = (e) => {
		setValue(e.target.value);
	}

	/**
	 * Función del form realizada para que no se actualice el form automáticamente.
	 */
	const subirInfoManualmente = (e) => e.preventDefault();

	/**
	 * Función que le dará al botón la funcionalidad de agregar la categoría
	 */
	const agregaCategoria = () => {

		if( value.trim().length > 1) {
			setCategorias([value, ...categorias]);
			setValue('');
		}
	}

    return (
        <form onSubmit={subirInfoManualmente}>
        	<h3> Agregar Categoria </h3>
        	<input 
				type='text' 
				value={value}
				placeholder='Escribe una categoría a agregar'
				onChange={cambiarValor}
			/>
			<div className='contenedor'>
				<div className="boton__agrega">
					<button onClick={agregaCategoria}>
            		    <span>Agregar</span>
            		</button>
				</div>
			</div> 
        </form>
    )
}

AgregarCategoria.propTypes = {
	categorias: PropTypes.array.isRequired,
	setCategorias: PropTypes.func.isRequired
};
