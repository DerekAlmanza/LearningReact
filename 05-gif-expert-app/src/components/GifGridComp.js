import React, { useState, useEffect } from 'react'

export const GifGridComp = ( {categoria} ) => {

    // const [contador, setContador] = useState(0);

    // UseEffect sirve para ejecutar cÃ³digo a partir de una condicional
    useEffect(() => {
        obtenerGif();
    },[]);

    /**
     * Obtener gif por medio de peticion https usando fetch
     */
    const obtenerGif = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=matrix&limit=10&api_key=3YLDhXWv1S63xr8KcyczSPq3D3Xh4Ftb';
        const res = await fetch( url );
        const { data } = await res.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images.original.url
            }
        })

        console.log(gifs);
    }

    return (
        <>
            <h3> {categoria} </h3>
        </>
    )
}
