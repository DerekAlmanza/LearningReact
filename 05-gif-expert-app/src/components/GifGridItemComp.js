import React from 'react'

export const GifGridItemComp = ( {id, title, url}) => {

    console.log( {id, title, url });

    return (
        <div className="contenedor-imagen">
            <div>
                <img className="imagen__estilo" src={ url } alt={title} />
                <p className="titulo-imagen__estilo"> {title} </p>
            </div>
        </div>
    )
}
