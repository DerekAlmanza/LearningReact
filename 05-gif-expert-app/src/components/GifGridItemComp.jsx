import React from 'react';
import PropTypes  from 'prop-types';

export const GifGridItemComp = ( {id, title, url}) => {

    return (    
        <div className="contenedor-imagen animate__animated animate__backInDown">
            <div>
                <img className="imagen__estilo" src={ url } alt={title} />
                <p className="titulo-imagen__estilo"> {title} </p>
            </div>
        </div>
    )
}

GifGridItemComp.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};