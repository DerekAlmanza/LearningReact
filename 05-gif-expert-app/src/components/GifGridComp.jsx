import React, { useState, useEffect } from 'react'
import { GifGridItemComp } from './GifGridItemComp';
import { obtenerGif } from '../helpers/obtenerGif';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGridComp = ( {categoria} ) => {

    const [imagenes, setImagenes] = useState([]);

    const {data, loading} = useFetchGifs(categoria);

    // UseEffect sirve para ejecutar código a partir de una condicional
    useEffect(() => {
        obtenerGif( categoria )
            .then(imgs => setImagenes(imgs))
            .catch(err => console.log(err));
    },[ categoria ]); // Por si llegara a cambiar la categoría.

    return (
        <>
            <h3> {categoria.toUpperCase()} </h3>
            {loading && <p className='animate__animated animate__flash'>Loading</p>}
            <div className="organiza-items">
                {
                    imagenes.map((imagen) => {
                        return (
                            <GifGridItemComp key={imagen.id} {...imagen} />
                        )
                    })
                }
            </div>
        </>
    )
}

GifGridComp.propTypes = {
    categoria: PropTypes.string.isRequired
};
