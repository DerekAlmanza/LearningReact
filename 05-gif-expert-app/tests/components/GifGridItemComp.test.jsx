import { render, screen } from "@testing-library/react";
import React from "react";
import { GifGridItemComp } from '../../src/components/GifGridItemComp';

describe('Pruebas en <GifGridItemComp />', () => {

    const title = "matrix";
    const url = "https://matrix.com/codigo.jpg";

    test('Debe de hacer match con el Snapshot ', () => {

        const {contenedor} = render(<GifGridItemComp title={title} url={url}/>);

        expect(contenedor).toMatchSnapshot();
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render(<GifGridItemComp title={title} url={url}/>);
        //screen.debug(); //an√°logo a un console.log() pero para mostrarlo en tests.
        console.log( screen.findAllByRole('img'));
    })
});