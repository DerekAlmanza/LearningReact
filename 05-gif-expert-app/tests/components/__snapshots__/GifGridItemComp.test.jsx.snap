// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItemComp /> Debe de hacer match con el Snapshot  1`] = `undefined`;
